src/util.ts

```diff
@@ -29,3 +30,51 @@ export function convertToUint8Array(data: string): Uint8Array {
 export function retryDeadline<T>(t: Promise<T>, timeout: number): Promise<T> {
   return retry<T>(() => deadline(t, timeout));
 }
+
+/**
+ * Utility to convert a CDP network request to a `Request` object
+ * Note: the received body is encoded in base64
+ */
+export function cdpRequestToRequest(request: Network_Request): Request {
+  const encoder = new TextEncoder();
+  const body = request.hasPostData
+    ? new ReadableStream<Uint8Array<ArrayBufferLike>>({
+      pull(controller) {
+        const entry = request.postDataEntries!.shift();
+        if (!entry?.bytes) {
+          return controller.close();
+        }
+        controller.enqueue(encoder.encode(atob(entry.bytes)));
```

Comment:
atob and btoa has a few issues with arbitrary binary data iirc, please use std's decoding instead of using them directly.